packages/astro/components/ViewTransitions.astro

```diff
@@ -194,5 +207,14 @@ const { fallback = 'animate' } = Astro.props as Props;
 			);
 		});
 		addEventListener('load', onload);
+		// There's not a good way to record scroll position before a back button.
+		// So the way we do it is by listening to scroll and just continuously recording it.
+		addEventListener('scroll', () => {
+			const state: State | undefined = history.state;
+			if(state) {
+				state.scrollY = scrollY;
+				history.replaceState(state, '');
+			}
+		}, { passive: true });
```

Comment:
Even though with passive, I wonder if there's a perf penalty calling `replaceState` repeatedly. Maybe it can use a throttle.

Could keep an eye on this and leave it as is too now though.

Reply:
I agree with Bjorn, but I feel more strongly on this one. I think **we have** to add a bouncing mechanism.

Reply:
If it were a regular Map would you still feel this way? I would assume that `replaceState` is just a map under the hood. I don't want to add extra code, increasing the bundle size, unless we know that we need to.

Reply:
Given that this is an experimental feature I think it's ok to be less safe in this case. If it is a problem we'll hear about it and fix before it becomes stable.

Reply:
Did some investigation and Safari considers calling replaceState too many times a security issue. I think I understand why; malicious sites could call pushState a lot to prevent you from backing out of the site. But I don't see why that would be the case for replaceState.

In any event, will need to look for another solution here.

Reply:
I remember very vaguely hitting this issue, and was able to fix it with a `throttle` helper with 300ms timeout (i think). This SO post seems to explain it: https://stackoverflow.com/questions/38419867/os-x-ios-safari-history-replacestate-limit-throws-securityerror-dom-exception

Reply:
Yeah, that's what I wound up doing. I did 500ms to be safe, but can bump it down. Otherwise ready for another review.